Mathlib/ModelTheory/ElementaryMaps.lean

```diff
@@ -106,12 +106,7 @@ theorem elementarilyEquivalent (f : M ↪ₑ[L] N) : M ≅[L] N :=
 @[simp]
 theorem injective (φ : M ↪ₑ[L] N) : Function.Injective φ := by
   intro x y
-  have h :=
-    φ.map_formula ((var 0).equal (var 1) : L.Formula (Fin 2)) fun i => if i = 0 then x else y
-  rw [Formula.realize_equal, Formula.realize_equal] at h
-  simp only [Term.realize, Fin.one_eq_zero_iff, if_true,
-    Function.comp_apply] at h
-  exact h.1
+  exact (φ.map_formula ((var 0).equal (var 1)) fun i => if i = 0 then x else y).1
```

Comment:
You can go one step further:
```suggestion
theorem injective (φ : M ↪ₑ[L] N) : Function.Injective φ :=
  fun x y ↦ (φ.map_formula ((var 0).equal (var 1)) fun i ↦ if i = 0 then x else y).1
```
Also, can any of the parts of this proof be inferred? (like the `fun i ↦ if i = 0 then x else y` term). Otherwise LGTM.